qoffer-domain/src/main/groovy/life/qbic/portal/portlet/offers/update/UpdateOfferDataSource.groovy

```diff
@@ -0,0 +1,30 @@
+package life.qbic.portal.portlet.offers.update
+
+import life.qbic.datamodel.dtos.business.Offer
+import life.qbic.portal.portlet.exceptions.DatabaseQueryException
+
+/**
+ * Provides methods to store offers in a data-source
+ *
+ * @since: 1.0.0
+ */
+interface UpdateOfferDataSource {
+
+    /**
+     * Saves an offer in a persistent data-source.
+     *
+     * Developers should execute this method, when they
+     * want to store an offer in f.e. a database.
+     *
+     * If the passed offer is already in the database (offer
+     * content with the same offer identifier already exists),
+     * then this method will throw a DatabaseQueryException.
+     *
+     * If the passed offer cannot be saved successfully, this method
+     * will also throw a DatabaseQueryException.
+     *
+     * @param offer
+     * @throws life.qbic.portal.portlet.exceptions.DatabaseQueryException
+     */
+    void store(Offer offer) throws DatabaseQueryException
```

Comment:
```suggestion
    void update(Offer offer) throws DatabaseQueryException
```

or 

```suggestion
    void store(Offer updatedOffer) throws DatabaseQueryException
```

to make it more explicit. Otherwise it is the same as the `CreateOfferDataSource` and thus be duplication.

Reply:
but it is the same and does the same.

Reply:
Ok, but then I don't understand why you do not reuse the data source that is already there to store offers. This is the boilercode imo.

Reply:
I do reuse the data source implementation. Just not the interface as a different use case requires a different interface.   I would opt to move the storing out of both use cases tbh.